build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -244,41 +244,32 @@ export const handlers: Handler[] = [
 			return undefined;
 		},
 		resolver: (file) => {
-			let json;
-			try {
-				json = JSON.parse(readFile(file));
-			} catch (err) {
-				return { resolved: false, message: "Error parsing JSON file: " + file };
-			}
-
-			const resolved = true;
-
-			if (json.author === undefined || json.author !== author) {
-				json.author = author;
-			}
-
-			if (json.license === undefined || json.license !== licenseId) {
-				json.license = licenseId;
-			}
+			updatePackageJsonFile(path.dirname(file), (json) => {
+				if (json.author === undefined || json.author !== author) {
+					json.author = author;
+				}
 
-			if (json.repository === undefined || typeof json.repository === "string") {
-				json.repository = {
-					type: "git",
-					url: repository,
-				};
-			}
+				if (json.license === undefined || json.license !== licenseId) {
+					json.license = licenseId;
+				}
 
-			if (json.repository.url !== repository) {
-				json.repository.url = repository;
-			}
+				if (json.repository === undefined || typeof json.repository === "string") {
+					json.repository = {
+						type: "git",
+						url: repository,
+					};
+				}
 
-			if (json.homepage === undefined || json.homepage !== homepage) {
-				json.homepage = homepage;
-			}
+				if (json.repository.url !== repository) {
+					json.repository.url = repository;
+				}
```

Comment:
Is there a reason not to just set this unconditionally?
```suggestion
				json.repository = {
					type: "git",
					url: repository,
				};
```

Reply:
ah, I see there is also a directory field. Can you just set this unconditionally and include the directory field then? I'll update the suggestion. Then maybe remove the whole check below which lints for the directory?

Reply:
I cleaned this up and ended up removing the resolver for npm-package-json-lint since it was duplicating this.